server/src/expense/expense.service.ts

```diff
@@ -139,61 +136,165 @@ export class ExpenseService {
         );
       }
 
-      const firstDayOfMonth = new Date(year, month - 1, 1);
-      const lastDayOfMonth = new Date(year, month, 0);
+      const expensesPerMonth = [];
 
-      const expenses = await this.prisma.expense.findMany({
-        where: {
-          houseId: userHouse.houseId,
-          date: {
-            gte: firstDayOfMonth,
-            lte: lastDayOfMonth,
+      // Iterate over the last three months, including the current month
+      for (let i = 0; i <= 2; i++) {
```

Comment:
If some part is the same logic as `getExpensePerDate`, you can make a reusable function!

Reply:
Fixed!
https://github.com/HAK2024/fairy-share/pull/119/commits/b3c1c2ec8b665e5b61a11b24d95470a37b82a13e